### ISSUE TITLE
get_synonyms.py fails to handle mouse data

### ISSUE THREAD

**Issue by xc611** [ndaniel/fusioncatcher#51]
Hi,

I tried to download mouse data by using fusioncatcher-build;

fusioncatcher-build -g mus_musculus -o .

    however I got error:
    
    ////////////////////////////////////////////////////////////////////////////////
      Running: step = 9   Time: 10:48   Date: 2017-06-08 (elapsed time: 0d:0h:8m)
    \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
    ==> Current working directory: '/gpfs/gsfs4/users/CCBR/projects/ccbr757/fc'
    get_synonyms.py \
    --organism mus_musculus \
    --server ftp.ensembl.org \
    --output /gpfs/gsfs4/users/CCBR/projects/ccbr757/fc/
    --------------------------------------------------------------------------------
    +-->EXECUTING...
    Downloading the MySQL files of organism 'mus_musculus' from Ensembl!
    230 Login successful.
    ERROR: '/pub/current_mysql/mus_musculus_core_' not found!

The reason is there are more than 1 item started with mus_musculus_core_89_38

I fixed line 139 to:

if organism_dir and len(organism_dir) >= 1:

Then there is another bug at line 192, I added "and parts[0].find('Table structure') == -1"
However error KeyError: None is still exists since last_name is empty.

I don't want to change your code too much, please take a look this error.

Thanks

Jack

**Comment by ndaniel** [comment]
Hi Jack,

what version of FusionCatcher do you have there?

Cheers,
Daniel

**Comment by ndaniel** [comment]
It looks that this is a temporary fix but it needs more testing.

Change line 138 in `get_synonyms.py` from
```
        organism_dir = [el for el in ftp.nlst() if el.lower().startswith(key) ]
```
to
```
        organism_dir = [el for el in ftp.nlst() if el.lower().startswith(key) and not el.endswith('.gz')]
```

**Comment by xc611** [comment]
Thanks, I will give a try.

Jack

**Comment by xc611** [comment]
Hi, Daniel:

Wolfgang helped me to try out your fix. However the KeyError: None is still exists since last_name is empty.

////////////////////////////////////////////////////////////////////////////////
   Running: step = 9   Time: 09:08   Date: 2017-06-12 (elapsed time: 0d:0h:25m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/spin1/users/wresch/temp'
get_synonyms.py \
--organism mus_musculus \
--server ftp.ensembl.org \
--output /spin1/users/wresch/temp/mm/
--------------------------------------------------------------------------------
+-->EXECUTING...
Downloading the MySQL files of organism 'mus_musculus' from Ensembl!
230 Login successful.
Downloading: ftp.ensembl.org/pub/current_mysql/mus_musculus_core_89_38/object_xref.txt.gz
Downloading: ftp.ensembl.org/pub/current_mysql/mus_musculus_core_89_38/xref.txt.gz
Downloading: ftp.ensembl.org/pub/current_mysql/mus_musculus_core_89_38/external_synonym.txt.gz
Downloading: ftp.ensembl.org/pub/current_mysql/mus_musculus_core_89_38/mus_musculus_core_89_38.sql.gz
Downloading: ftp.ensembl.org/pub/current_mysql/mus_musculus_core_89_38/gene.txt.gz
Decompressing files ...
Parsing the tables in the SQL file...
Traceback (most recent call last):
   File "/usr/local/apps/fusioncatcher/0.99.7b/fusioncatcher/bin/get_synonyms.py", line 195, in <module>
     table[last_name][col] = idx
KeyError: None


ERROR: Workflow execution failed at step 9 while executing:
----------------
    get_synonyms.py \
    --organism mus_musculus \
    --server ftp.ensembl.org \
    --output /spin1/users/wresch/temp/mm/
----------------

**Comment by ndaniel** [comment]
Using this [get_synonyms.py](https://github.com/ndaniel/fusioncatcher/blob/master/bin/get_synonyms.py), does it fix it?

**Comment by xc611** [comment]
Yes, it worked fine this time. Thanks.

Jack

**Comment by ndaniel** [comment]
Ok. Thanks! This will be fixed in v0.99.7c!
